src/palimpzest/query/operators/rag.py

```diff
@@ -108,11 +108,11 @@ def compute_embedding(self, text: str) -> tuple[list[float], GenerationStats]:
 
         # compute the generation stats object
         model_card = MODEL_CARDS[model_name]
-        total_input_tokens = response.usage.total_tokens
-        total_input_cost = model_card["usd_per_input_token"] * total_input_tokens
+        total_embedding_input_tokens = response.usage.total_embedding_input_tokens
+        total_input_cost = model_card["usd_per_input_token"] * total_embedding_input_tokens
         embed_stats = GenerationStats(
             model_name=model_name,  # NOTE: this should be overwritten by generation model in convert()
-            total_input_tokens=total_input_tokens,
+            total_input_tokens=total_embedding_input_tokens,
```

Comment:
Remember that we created the new fields in the `GenerationStats` object to track the cost of computing embeddings. As such, what we want here is to assign:
```
embed_stats = GenerationStats(
    model_name=model_name,
    total_input_tokens=0.0,
    total_output_tokens=0.0,
    total_input_cost=0.0,
    total_output_cost=0.0,
    total_embedding_input_tokens=total_embedding_input_tokens,
    total_embedding_cost=total_embedding_cost,
    cost_per_record=total_embedding_cost,  # using renamed variable from previous comment
    ...
)
    
```